src/otaclient/_status_monitor.py

```diff
@@ -309,32 +307,36 @@ def _status_collector_thread(self) -> None:
         """Main entry of status monitor working thread."""
         _next_shm_push = 0
         while True:
-            _now = time.time()
+            _now = time.perf_counter()
+
+            # ------ try process the report ------ #
+            report = None
             try:
                 report = self._input_queue.get_nowait()
                 if report is TERMINATE_SENTINEL:
-                    break
-
-                # ------ push status on load_report ------ #
-                # NOTE: always push OTAStatus change report
-                if (
-                    self.load_report(report)
-                    and self._status
-                    and (
-                        isinstance(report.payload, OTAStatusChangeReport)
-                        or _now > _next_shm_push
-                    )
-                ):
-                    try:
-                        self._shm_status.write_msg(self._status)
-                        _next_shm_push = _now + self.shm_push_interval
-                    except Exception as e:
-                        burst_suppressed_logger.debug(
-                            f"failed to push status to shm: {e!r}"
-                        )
+                    logger.info("status collector thread exits on receiving sentinal")
+                    return
+
+                # report that is invalid
+                if not self.load_report(report):
+                    report = None
             except queue.Empty:
                 time.sleep(self.min_collect_interval)
```

Comment:
When queue is empty, should we continue? Maybe same status might be written?

Reply:
( While for `OTAStatusChangeReport`, we still always push no matter the push interval.